src/puppetlabs/puppetdb/mq.clj

```diff
@@ -264,6 +266,45 @@
     (vec (repeatedly n #(commit-or-rollback s
                           (convert-jms-message (.receive consumer)))))))
 
+(defn- broker-mb
+  [name]
+  (str "org.apache.activemq:type=Broker,brokerName=" name))
+
+(defn-validated queue-size
+  "Returns the number of pending messages in the named broker queue.
+  Throws {:type ::queue-not-found :broker broker :name name} when the
+  queue doesn't exist."
+  [broker :- s/Str
+   queue :- s/Str]
+  (try
+    (jmx/read (format "%s,destinationType=Queue,destinationName=%s"
+                      (broker-mb broker) queue)
+              :QueueSize)
+    (catch javax.management.InstanceNotFoundException ex
+      (throw+ {:type ::queue-not-found :broker broker :name queue}))))
+
+(defn-validated transfer-messages!
+  "Transfers all of the messages currently available in the named
+  source queue to the destination and returns the number transferred.
+  Each message successfully transferred will be committed."
+  [broker :- s/Str
+   source :- s/Str
+   destination :- s/Str]
+  (jmx/invoke-signature (format "%s,destinationType=Queue,destinationName=%s"
+                                (broker-mb broker) source)
+                        :moveMatchingMessagesTo
+                        ["java.lang.String" "java.lang.String"]
+                        "JMSMessageID is NOT NULL"
+                        destination))
+
+(defn-validated remove-queue!
+  "Deletes the named broker queue.  Does nothing if the queue doesn't exist."
+  [broker :- s/Str
+   queue :- s/Str]
+  ;; When the queue doesn't exist, removeQueue provides no indication.
```

Comment:
Practically, I suppose that's correct, though I'd left it because I had to figure it out by testing, and thought it might save someone else time (i.e. so they don't think it's possible, but we just chose to ignore the return value or something).  In any case, I'm happy to remove it.